File: src/endpoints/auth/delete_item.py
from . import __types__ as types
from src import output
from src.libs.db import DB, Table
from src.modules.verify import checking_token

async def delete_item(params: types.DeleteItem, req = None, db: DB = None):

    # set values
    result = None
    db = db if db and isinstance(db, DB) else DB().connect()

    try:
        # checking token
        checking_token(req, db)

        # check data
        count = db.get_count(
            table_name = Table.PROVIDER.value,
            where = [ f'srl = {params.srl}' ],
        )
        if not (count > 0): raise Exception('Item not found.', 204)

        # delete data
        db.delete_item(
            table_name = Table.PROVIDER.value,
            where = [ f'srl = {params.srl}' ],
        )

        # delete tokens
        db.delete_item(
            table_name = Table.TOKEN.value,
            where = [ f'provider_srl = {params.srl}' ],
        )

        # set result
        result = output.success({
            'message': 'Complete delete provider.',
        })
    except Exception as e:
        result = output.exc(e)
    finally:
        if db: db.disconnect()
        return result

File: src/routes/category/delete-item.ts
import DB, { db } from '@/classes/DB'
import ServiceError from '@/classes/ServiceError'
import * as helper from './__helper'

export default async function deleteItem(srl: number)
{
  try
  {
    // check data
    const count = helper.count({
      where: `srl = ${srl}`,
    })
    if (count <= 0) throw new ServiceError('No data.', { status: 204 })

    // delete data
    db.deleteData({
      table: DB.TABLE.CATEGORY,
      where: `srl = ${srl}`,
    })
  }
  catch (_e: any)
  {
    throw new ServiceError('Failed to delete Category.', {
      status: _e.status,
      text: _e.message,
      cause: _e,
    })
  }
}
